### ISSUE TITLE
Better backwards compatibility with optional transformers that are now plugins

### ISSUE THREAD

**Issue by aickin** [babel/babel#1995]
I've been having a lot of difficulty getting Babel plugins to work consistently. I think I have a reproducible test case here:

```
curl https://dl.dropboxusercontent.com/u/36665376/babel-plugin-test.tar.gz > babel-plugin-test.tar.gz
tar -zxvf babel-plugin-test.tar.gz
cd babel-plugin-test
npm install
node build.js
```

On my machine this gives the following error:

```
/Users/xxx/code/babel-plugin-test/node_modules/babel-core/lib/babel/transformation/file/plugin-manager.js:158
      throw new ReferenceError(messages.get("pluginKeyCollision", key));
            ^
ReferenceError: The plugin "minification.removeConsole" collides with another of the same name
    at PluginManager.validate (/Users/xxx/code/babel-plugin-test/node_modules/babel-core/lib/babel/transformation/file/plugin-manager.js:158:13)
    at PluginManager.add (/Users/xxx/code/babel-plugin-test/node_modules/babel-core/lib/babel/transformation/file/plugin-manager.js:212:10)
    at File.buildTransformers (/Users/xxx/code/babel-plugin-test/node_modules/babel-core/lib/babel/transformation/file/index.js:235:21)
    at new File (/Users/xxx/code/babel-plugin-test/node_modules/babel-core/lib/babel/transformation/file/index.js:135:10)
    at Pipeline.transform (/Users/xxx/code/babel-plugin-test/node_modules/babel-core/lib/babel/transformation/pipeline.js:167:16)
    at Object.<anonymous> (/Users/xxx/code/babel-plugin-test/build.js:1:96)
    at Module._compile (module.js:460:26)
    at Object.Module._extensions..js (module.js:478:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
```

It's worth noting that I have been able to get plugins to work once in a while, but they soon start giving this error, but I haven't pinned down why and when it breaks. I apologize if there's something odd that I've done with my config, and hopefully this test will repro for you.

node: v0.12.7
npm: v2.12.1
OSX: v10.10.3

**Comment by sebmck** [comment]
I assume you're trying to use plugins from the [babel-plugins](http://github.com/babel-plugins) org?

**Comment by aickin** [comment]
Yes. I've only tried ones from babel-plugins. The test project uses `babel-plugin-remove-console`, but I've repro-ed at various times with several other plugins.

**Comment by sebmck** [comment]
Yeah, that's because those plugins are technically still integrated inside of Babel itself for backwards compat reasons. So instead of using:

```sh
$ babel --plugins remove-console
```

you can use:

```sh
$ babel --optional minification.removeConsole
```

and it should work without any issues.

**Comment by aickin** [comment]
Ah, thanks for the tip! I tried {optional: ["remove-console"]}, but that did not work.

Am I right to think that the documentation in https://github.com/babel-plugins/babel-plugin-remove-console/blob/master/README.md is incorrect, then?

Going off to try using `optional` syntax; I'll report back in a few minutes.

**Comment by sebmck** [comment]
@aickin Yeah the `optional` names are different since they reference the names of the internal transformers, they're documented [here](http://babeljs.io/docs/advanced/transformers/#optional).

**Comment by aickin** [comment]
Cool, and this fixes my test case.

Mind if I make some PRs for the READMEs for some babel-plugins? It seems like most of them tell you incorrectly to use `plugins` rather than `optional`.

**Comment by sebmck** [comment]
I should probably fix this internally actually. A deprecation message needs to be output to get users to start using the plugins (which apparently don't work when used directly) rather than using it internally.

**Comment by aickin** [comment]
OK. LMK if I can help, and thanks for the quick response and all the great work you do. (Saw your talk at React Europe the other week and was very impressed.)

I'll leave this issue open since you want to fix it internally, but using `optional` completely unblocks me. Thanks!

**Comment by sebmck** [comment]
No problem, thanks for the report! Will rename and leave this open until it's fixed.

**Comment by thejameskyle** [comment]
I assume this is no longer relevant with 6.0. Closing, feel free to reopen.
